diff --git a/python/make_source.sh b/python/make_source.sh
new file mode 100755
index 0000000..4fcc414
--- /dev/null
+++ b/python/make_source.sh
@@ -0,0 +1,36 @@
+#!/bin/bash -xe
+# Copyright 2022 DeepMind Technologies Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Figure out the path to this script (https://stackoverflow.com/a/246128).
+package_dir="$(cd -- "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 && pwd)"
+
+export PYTHONPATH="${package_dir}"/mujoco
+python "${package_dir}"/mujoco/codegen/generate_enum_traits.py > \
+    mujoco/enum_traits.h
+python "${package_dir}"/mujoco/codegen/generate_function_traits.py > \
+    mujoco/function_traits.h
+python "${package_dir}"/mujoco/codegen/generate_spec_bindings.py > \
+    mujoco/specs.cc.inc
+export PYTHONPATH=
+
+# Copy over the LICENSE file.
+cp "${package_dir}"/../LICENSE .
+
+# Copy over CMake scripts.
+mkdir mujoco/cmake
+cp "${package_dir}"/../cmake/*.cmake mujoco/cmake
+
+# Copy over Simulate source code.
+cp -r "${package_dir}"/../simulate mujoco
