diff --git a/simulate/CMakeLists.txt b/simulate/CMakeLists.txt
index 97b10f3..8b3ebaa 100644
--- a/simulate/CMakeLists.txt
+++ b/simulate/CMakeLists.txt
@@ -80,26 +80,13 @@ if(MUJOCO_HARDEN)
   endif()
 endif()
 
-# Fetch lodepng dependency.
-if(NOT TARGET lodepng)
-  FetchContent_Declare(
-    lodepng
-    GIT_REPOSITORY https://github.com/lvandeve/lodepng.git
-    GIT_TAG ${MUJOCO_DEP_VERSION_lodepng}
-  )
-
-  FetchContent_GetProperties(lodepng)
-  if(NOT lodepng_POPULATED)
-    FetchContent_Populate(lodepng)
-    # This is not a CMake project.
-    set(LODEPNG_SRCS ${lodepng_SOURCE_DIR}/lodepng.cpp)
-    set(LODEPNG_HEADERS ${lodepng_SOURCE_DIR}/lodepng.h)
-    add_library(lodepng OBJECT ${LODEPNG_HEADERS} ${LODEPNG_SRCS})
-    target_compile_options(lodepng PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
-    target_link_options(lodepng PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
-    target_include_directories(lodepng PUBLIC ${lodepng_SOURCE_DIR})
-  endif()
-endif()
+set(lodepng_SOURCE_DIR ${CMAKE_SOURCE_DIR}/third_party/lodepng)
+set(LODEPNG_SRCS ${lodepng_SOURCE_DIR}/lodepng.cpp)
+set(LODEPNG_HEADERS ${lodepng_SOURCE_DIR}/lodepng.h)
+add_library(lodepng OBJECT ${LODEPNG_HEADERS} ${LODEPNG_SRCS})
+target_compile_options(lodepng PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
+target_link_options(lodepng PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
+target_include_directories(lodepng PUBLIC ${lodepng_SOURCE_DIR})
 
 # Simulate library
 add_library(platform_ui_adapter OBJECT)
