diff --git a/onnxruntime/python/onnxruntime_pybind_state.cc b/onnxruntime/python/onnxruntime_pybind_state.cc
index 9a9243e..d7ed7e6 100644
--- a/onnxruntime/python/onnxruntime_pybind_state.cc
+++ b/onnxruntime/python/onnxruntime_pybind_state.cc
@@ -2378,8 +2520,8 @@ class EnvInitializer {
     Env::Default().GetTelemetryProvider().SetLanguageProjection(OrtLanguageProjection::ORT_PROJECTION_PYTHON);
     OrtPybindThrowIfError(Environment::Create(std::make_unique<LoggingManager>(
                                                   std::make_unique<CLogSink>(),
-                                                  Severity::kWARNING, false, LoggingManager::InstanceType::Default,
-                                                  &SessionObjectInitializer::default_logger_id),
+                                                  Severity::kWARNING, false, LoggingManager::InstanceType::Temporal,
+                                                  nullptr),
                                               env_ptr));
     session_env_ = std::shared_ptr<Environment>(env_ptr.release());
     destroyed = false;
