diff --git a/setup.py b/setup.py
index 51feedcfd3..258b07e1d8 100644
--- a/setup.py
+++ b/setup.py
@@ -599,6 +599,9 @@ if enable_training or enable_training_apis:
 if package_name == "onnxruntime-tvm":
     packages += ["onnxruntime.providers.tvm"]
 
+# TODO: introduce onnxruntime-et package
+packages += ["onnxruntime.providers.etglow"]
+
 package_data["onnxruntime"] = data + examples + extra
 
 version_number = ""
diff --git a/tools/ci_build/build.py b/tools/ci_build/build.py
index 587d035541..0a1805a566 100644
--- a/tools/ci_build/build.py
+++ b/tools/ci_build/build.py
@@ -756,6 +756,7 @@ def parse_arguments():
     parser.add_argument("--use_xnnpack", action="store_true", help="Enable xnnpack EP.")
     parser.add_argument("--use_avx512", action="store_true", help="Enable AVX512 instructions")
     parser.add_argument("--use_azure", action="store_true", help="Enable azure EP.")
+    parser.add_argument("--use_etglow", action="store_true", help="Enable ETGLOW EP.")
 
     parser.add_argument("--use_cache", action="store_true", help="Use compiler cache in CI")
 
@@ -1092,6 +1093,7 @@ def generate_build_tree(
         "-Donnxruntime_DISABLE_SPARSE_TENSORS=" + ("ON" if disable_sparse_tensors else "OFF"),
         "-Donnxruntime_DISABLE_OPTIONAL_TYPE=" + ("ON" if disable_optional_type else "OFF"),
         "-Donnxruntime_CUDA_MINIMAL=" + ("ON" if args.enable_cuda_minimal_build else "OFF"),
+        "-Donnxruntime_USE_ETGLOW=" + ("ON" if args.use_etglow else "OFF"),
     ]
 
     if args.rv64:
