diff --git a/src/python/py/models/builder.py b/src/python/py/models/builder.py
index 1ef5ea78..357443a8 100644
--- a/src/python/py/models/builder.py
+++ b/src/python/py/models/builder.py
@@ -75,6 +75,7 @@ class Model:
             "dml": {},
             "webgpu": {},
             "NvTensorRtRtx": {},
+            "ETGLOW": {},
         }
 
         # Map input names to their types and shapes
@@ -344,6 +345,7 @@ class Model:
             ("webgpu", TensorProto.FLOAT16),
             ("webgpu", TensorProto.FLOAT),
             ("NvTensorRtRtx", TensorProto.FLOAT16),
+            ("ETGLOW", TensorProto.FLOAT16),
         ]
         if (self.ep, self.io_dtype) in valid_gqa_configurations:
             # Change model settings for GroupQueryAttention
@@ -3850,7 +3852,7 @@ def get_args():
         "-e",
         "--execution_provider",
         required=True,
-        choices=["cpu", "cuda", "rocm", "dml", "webgpu", "NvTensorRtRtx"],
+        choices=["cpu", "cuda", "rocm", "dml", "webgpu", "NvTensorRtRtx", "ETGLOW"],
         help="Execution provider to target with precision of model (e.g. FP16 CUDA, INT4 CPU, INT4 WEBGPU)",
     )
 
