diff --git a/third_party/libccd/CMakeLists.txt b/third_party/libccd/CMakeLists.txt
index 35bf430..1d8ddf5 100644
--- a/third_party/libccd/CMakeLists.txt
+++ b/third_party/libccd/CMakeLists.txt
@@ -62,14 +62,14 @@ write_basic_package_version_file(ccd-config-version.cmake
   VERSION ${CCD_VERSION} COMPATIBILITY AnyNewerVersion)
 
 install(FILES
-  "${CMAKE_BINARY_DIR}/ccd-config.cmake"
-  "${CMAKE_BINARY_DIR}/ccd-config-version.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/ccd-config.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/ccd-config-version.cmake"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/ccd")
 
 set(CCD_PKGCONFIG_DESCRIPTION
   "Library for collision detection between convex shapes")
 configure_file(ccd.pc.in ccd.pc @ONLY)
-install(FILES "${CMAKE_BINARY_DIR}/ccd.pc"
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/ccd.pc"
   DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
 install(FILES BSD-LICENSE DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/doc/ccd")
