diff --git a/torch/lib/libshm/CMakeLists.txt b/torch/lib/libshm/CMakeLists.txt
index c3cd26f..911e5d4 100644
--- a/torch/lib/libshm/CMakeLists.txt
+++ b/torch/lib/libshm/CMakeLists.txt
@@ -64,7 +64,7 @@ if(BUILD_LIBTORCHLESS)
   target_link_libraries(torch_shm_manager PRIVATE shm ${C10_LIB})
 else()
   # we need to link directly to c10 here otherwise we miss symbols
-  target_link_libraries(torch_shm_manager PRIVATE shm c10)
+  target_link_libraries(torch_shm_manager PRIVATE shm c10 onnx onnx_proto protobuf::libprotobuf)
 endif()
 set_target_properties(torch_shm_manager PROPERTIES
   INSTALL_RPATH "${_rpath_portable_origin}/../lib")
