diff --git a/CMakeLists.txt b/CMakeLists.txt
index b666eec4..894776fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -104,10 +104,7 @@ endif()
 # find_package Python has replaced PythonInterp and PythonLibs since cmake 3.12
 # Use the following command in the future; now this is only compatible with the latest pybind11
 # find_package(Python ${PY_VERSION} COMPONENTS Interpreter Development REQUIRED)
-find_package(PythonInterp ${PY_VERSION} REQUIRED)
-if(BUILD_ONNX_PYTHON)
-  find_package(PythonLibs ${PY_VERSION})
-endif()
+find_package(Python ${PY_VERSION} COMPONENTS Interpreter Development REQUIRED)
 
 if(CMAKE_SYSTEM_NAME STREQUAL "AIX")
   set(CMAKE_NO_SYSTEM_FROM_IMPORTED 1)
@@ -361,7 +358,7 @@ function(RELATIVE_PROTOBUF_GENERATE_CPP NAME SRCS HDRS ROOT_DIR DEPEND)
     endif()
 
     add_custom_command(OUTPUT "${GENERATED_PROTO}"
-                       COMMAND "${PYTHON_EXECUTABLE}" "${GEN_PROTO_PY}"
+                       COMMAND "${Python_EXECUTABLE}" "${GEN_PROTO_PY}"
                                ARGS ${GEN_PROTO_ARGS}
                        DEPENDS ${INFILE}
                        COMMENT "Running gen_proto.py on ${INFILE}"
@@ -566,7 +563,7 @@ if(BUILD_ONNX_PYTHON)
 
   target_include_directories(onnx_cpp2py_export PUBLIC
     "${pybind11_INCLUDE_DIRS}"
-    "${PYTHON_INCLUDE_DIRS}")
+    "${Python_INCLUDE_DIRS}")
 
   if(APPLE)
     set_target_properties(onnx_cpp2py_export
@@ -600,7 +597,7 @@ if(BUILD_ONNX_PYTHON)
   target_link_libraries(onnx_cpp2py_export PRIVATE onnx)
 
   if(MSVC)
-    target_link_libraries(onnx_cpp2py_export PRIVATE ${PYTHON_LIBRARIES})
+    target_link_libraries(onnx_cpp2py_export PRIVATE ${Python_LIBRARIES})
     target_compile_options(onnx_cpp2py_export
                            PRIVATE /MP
                                    /wd4244 # 'argument': conversion from 'google::
diff --git a/tools/protoc-gen-mypy.sh.in b/tools/protoc-gen-mypy.sh.in
index 6231cb29..3b699915 100644
--- a/tools/protoc-gen-mypy.sh.in
+++ b/tools/protoc-gen-mypy.sh.in
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 # use PYTHON_EXECUTABLE from CMake to get the right python to execute
-"@PYTHON_EXECUTABLE@" -u "$DIR"/protoc-gen-mypy.py
+"@Python_EXECUTABLE@" -u "$DIR"/protoc-gen-mypy.py
