diff --git a/host/sw-sysemu/CMakeLists.txt b/host/sw-sysemu/CMakeLists.txt
index e306149..476e709 100644
--- a/host/sw-sysemu/CMakeLists.txt
+++ b/host/sw-sysemu/CMakeLists.txt
@@ -8,7 +8,7 @@
 # agreement/contract under which the program(s) have been supplied.
 #------------------------------------------------------------------------------
 
-cmake_minimum_required(VERSION 3.21)
+cmake_minimum_required(VERSION 3.26)
 project(sw-sysemu VERSION 0.21.0 LANGUAGES ASM C CXX)
 
 list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR})
@@ -71,7 +71,7 @@ endif()
 include(GNUInstallDirs)
 
 find_package(elfio REQUIRED)
-find_package(glog REQUIRED)
+find_package(glog 0.7.1 REQUIRED)
 find_package(lz4 REQUIRED)
 
 # dependencies
@@ -460,7 +460,6 @@ include(GenerateExportHeader)
 generate_export_header(sw-sysemu EXPORT_FILE_NAME ${SW_SYSEMU_PUBLIC_EXPORT_HEADER})
 target_sources(sw-sysemu PRIVATE ${SW_SYSEMU_PUBLIC_EXPORT_HEADER})
 target_link_libraries(sw-sysemu
-    PRIVATE
         elfio::elfio
         $<$<BOOL:${BACKTRACE}>:libunwind::libunwind>
         lz4::lz4
@@ -506,7 +505,7 @@ else()
     message(STATUS "Regular Build. Building sys_emu executable.")
 
     add_executable(sys_emu sw-sysemu/main.cpp)
-    target_link_libraries(sys_emu PRIVATE sw-sysemu)
+    target_link_libraries(sys_emu sw-sysemu)
     target_include_directories(sys_emu
         PRIVATE
             $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
diff --git a/host/sw-sysemu/agent.h b/host/sw-sysemu/agent.h
index 962d3b4..89e3dc6 100644
--- a/host/sw-sysemu/agent.h
+++ b/host/sw-sysemu/agent.h
@@ -12,6 +12,7 @@
 #define BEMU_AGENT_H
 
 #include <string>
+#include <cstdint>
 
 namespace bemu {
 
diff --git a/host/sw-sysemu/cmake/Findglog.cmake b/host/sw-sysemu/cmake/Findglog.cmake
deleted file mode 100644
index 35b8192..0000000
--- a/host/sw-sysemu/cmake/Findglog.cmake
+++ /dev/null
@@ -1,51 +0,0 @@
-# - Try to find Glog
-#
-# The following variables are optionally searched for defaults
-#  GLOG_ROOT_DIR:            Base directory where all GLOG components are found
-#
-# The following targets are available after glog is found:
-#  glog
-#  glog::glog
-
-include(FindPackageHandleStandardArgs)
-
-set(GLOG_ROOT_DIR "" CACHE PATH "Folder contains Google glog")
-
-if(WIN32)
-    find_path(GLOG_INCLUDE_DIR glog/logging.h
-        PATHS ${GLOG_ROOT_DIR}/src/windows)
-else()
-    find_path(GLOG_INCLUDE_DIR glog/logging.h
-        PATHS ${GLOG_ROOT_DIR})
-endif()
-
-if(MSVC)
-    find_library(GLOG_LIBRARY_RELEASE libglog_static
-        PATHS ${GLOG_ROOT_DIR}
-        PATH_SUFFIXES Release)
-
-    find_library(GLOG_LIBRARY_DEBUG libglog_static
-        PATHS ${GLOG_ROOT_DIR}
-        PATH_SUFFIXES Debug)
-
-    set(GLOG_LIBRARY optimized ${GLOG_LIBRARY_RELEASE} debug ${GLOG_LIBRARY_DEBUG})
-else()
-    find_library(GLOG_LIBRARY glog
-        PATHS ${GLOG_ROOT_DIR}
-        PATH_SUFFIXES lib lib64)
-endif()
-
-find_package_handle_standard_args(glog DEFAULT_MSG GLOG_INCLUDE_DIR GLOG_LIBRARY)
-
-if (GLOG_FOUND)
-    add_library(glog UNKNOWN IMPORTED GLOBAL)
-    add_library(glog::glog ALIAS glog)
-    target_include_directories(glog INTERFACE ${GLOG_INCLUDE_DIR})
-    set_target_properties(glog PROPERTIES IMPORTED_LOCATION ${GLOG_LIBRARY})
-
-    set(GLOG_INCLUDE_DIRS ${GLOG_INCLUDE_DIR})
-    set(GLOG_LIBRARIES ${GLOG_LIBRARY})
-    message(STATUS "Found glog    (include: ${GLOG_INCLUDE_DIR}, library: ${GLOG_LIBRARY})")
-    mark_as_advanced(GLOG_ROOT_DIR GLOG_LIBRARY_RELEASE GLOG_LIBRARY_DEBUG
-                                   GLOG_LIBRARY GLOG_INCLUDE_DIR)
-endif()
\ No newline at end of file
diff --git a/host/sw-sysemu/sw-sysemu/SysEmuImp.cpp b/host/sw-sysemu/sw-sysemu/SysEmuImp.cpp
index b42a684..ffff1fc 100644
--- a/host/sw-sysemu/sw-sysemu/SysEmuImp.cpp
+++ b/host/sw-sysemu/sw-sysemu/SysEmuImp.cpp
@@ -18,6 +18,9 @@
 #include <future>
 #include <mutex>
 #include <thread>
+
+extern int optind;
+
 using namespace emu;
 
 namespace {
diff --git a/host/sw-sysemu/sw-sysemu/include/sw-sysemu/SysEmuOptions.h b/host/sw-sysemu/sw-sysemu/include/sw-sysemu/SysEmuOptions.h
index 9e9f431..177886b 100644
--- a/host/sw-sysemu/sw-sysemu/include/sw-sysemu/SysEmuOptions.h
+++ b/host/sw-sysemu/sw-sysemu/include/sw-sysemu/SysEmuOptions.h
@@ -11,6 +11,7 @@
 #pragma once
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #pragma GCC visibility push(default)
 
